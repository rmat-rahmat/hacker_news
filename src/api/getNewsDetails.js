
import { timeoutPromise } from "./utilities";

//Fetch top stories from hacker news API

const getNewsDetails = (id) => {
    return new Promise((resolve, reject) => {
        timeoutPromise(10000, new Error('Timed Out!'),
            fetch(`https://hacker-news.firebaseio.com/v0/item/${id}.json`))
            .then((response) => response.json())
            .then((json) => {
                resolve(json);
            })
            .catch(error => {
                reject(error);
            });
    });
};
export default getNewsDetails;