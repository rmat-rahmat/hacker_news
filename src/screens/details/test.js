/**
 * @format
 */

import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import Details from '.';
import getNewsDetails from '../../api/getNewsDetails';

describe('Test Details screen', () => {

const mockData = {
  "by": "Tester",
  "descendants": 71,
  "id": 1,
  "kids": [2, 20],
  "score": 111,
  "time": 1175714200,
  "title": "This is test story",
  "type": "story",
  "url": "http://www.getdropbox.com/u/2/screencast.html"
}

const mockComment = [
  {
    "by": "tester1",
    "id": 2,
    "kids": [],
    "parent": 1,
    "text": "this is comment 1",
    "time": 1314211127,
    "type": "comment"
  },
  {
    "by": "tester2",
    "id": 20,
    "kids": [3],
    "parent": 1,
    "text": "this is comment 2",
    "time": 1314211127,
    "type": "comment"
  }
];


it('fetch Comment', async () => {
  const makeFetchResponse = value => ({ json: async () => mockComment[value] })
  const mockFetch = jest.fn()
    .mockReturnValueOnce(makeFetchResponse(0))
    .mockReturnValueOnce(makeFetchResponse(1))
  global.fetch = mockFetch

  getNewsDetails(0).then(x => {
    expect(x.id).toEqualTo(2);
  })
  getNewsDetails(1).then(x => {
    expect(x.id).toEqualTo(20);
  })
  expect(mockFetch).toHaveBeenCalledTimes(2)
})


it('renders Details correctly', () => {
  const tree = renderer.create(
    <Details data={mockData} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
});