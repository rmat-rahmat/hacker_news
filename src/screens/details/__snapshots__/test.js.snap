// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Details screen renders Details correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      size="large"
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
        }
      }
    >
      Loading...
    </Text>
  </View>
</View>
`;
